src/ucs/configure.m4

```diff
@@ -300,6 +300,21 @@ AS_IF([test "x$enable_builtin_memcpy" != xno],
 	  [AC_DEFINE([ENABLE_BUILTIN_MEMCPY], [0], [Enable builtin memcpy])]
   )
 
+#
+# Enable non temporal buffer transfer
+#
+AC_ARG_ENABLE([nt-buffer-transfer],
+	AS_HELP_STRING([--enable-nt-buffer-transfer],
+	               [Enable non temporal buffer transfer routine, default: YES]),
+	[],
+	[enable_nt_buffer_transfer=yes])
+
+AS_IF([test "x$enable_nt_buffer_transfer" != xno],
+	  [AS_MESSAGE([enabling non temporal buffer transfer])
+	   AC_DEFINE([ENABLE_NT_BUFFER_TRANSFER], [1], [Enable non temporal buffer transfer])],
+	  [AC_DEFINE([ENABLE_NT_BUFFER_TRANSFER], [0], [Enable non temporal buffer transfer])]
+  )
```

Comment:
i think we should try removing the compile time condition and make it runtime choice only

Reply:
This is done to mimic "builtin-memcpy" implementation.  We can remove this entirely, but I need "#if __AVX__" around the nt-buffer-transfer code and #if __LZCNT__ around "_lzcnt_u32()"

Reply:
let's keep checking AVX and LZCNT, but remove enable_nt_buffer_transfer. I think there is no reason to disable it in compile time (if have AVX and LZCNT)

Reply:
Yes. It can be always compiled in.